docs/testing.rst

```diff
@@ -0,0 +1,98 @@
+=======
+Testing best practices
+=======
+
+Test set up
+===========
+
+Doing a lot of work in the ``setUp`` call of a test class means that it will be run on every test. This
+quickly adds a lot of run time to the tests. Some things that can be easily moved to ``setUpClass`` are domain
+creation, user creation, or any other static models needed for the test.
+
+Furthermore, nose provides the ability to write setup for the entire package and/or module. More information
+on that can be found `here <http://pythontesting.net/framework/nose/nose-fixture-reference/#package>`_.
+
+Using SimpleTestCase
+====================
+
+The SimpleTestCase runs tests without a database. Many times this can be achieved through the use of the `mock
+library <http://www.voidspace.org.uk/python/mock/>`_. A good rule of thumb is to have 80% of your tests be unit
+tests that utilize ``SimpleTestCase``, and then 20% of your tests be integration tests that utilize the
+database and ``TestCase``.
+
+CommCareHQ also has some custom in mocking tools.
+
+    - `Fake Couch <https://github.com/dimagi/fakecouch>`_ - Fake implementation of CouchDBKit api for testing purposes.
+    - `ESQueryFake <https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/es/fake/es_query_fake.py>`_ - For faking ES queries.
+
+
+Tearing down tests
+==================
+
+It is important to ensure that all objects you have created in the test database are deleted when the test
+class finishes running. This often happens in the ``tearDown`` method or the ``tearDownClass`` method.
+However, unneccessary cleanup "just to be safe" can add a large amount of time onto your tests.
+
+Combining test classes
+======================
+
+Sometimes classes share the same base class and inherit the ``setUpClass`` function. Below is an example:
+
+.. code:: python
```

Comment:
These code samples are not showing up in the [rich text view](https://github.com/dimagi/commcare-hq/blob/4bed22abf2f22b582d9af0d4ba63cf1da31e9e08/docs/testing.rst)